### ISSUE TITLE
Table and column identifiers are no properly escaped.

### ISSUE THREAD

**Issue by javier-garcia-meteologica** [jawj/zapatos#98]

**Comment by jawj** [comment]
(1) Don't do this.
(2) To help you not do this, that query gives a type error, since the object passed in is not typed as a `Whereable`: see below.

<img width="1099" alt="Screenshot 2021-08-10 at 09 37 30" src="https://user-images.githubusercontent.com/141620/128835631-49a09b56-1ecf-468d-bad2-fdb5ceab8fcb.png">

**Comment by jawj** [comment]
OK, I see your pull request.

I need to remind myself why we check if a string is already quoted before we quote it there, since that does seem like a bit of a code smell, on the face of it.

But it remains the case that this quoting is not designed as a security measure, but only to protect JS-style camelCase identifiers from being downcased by Postgres.

**Comment by javier-garcia-meteologica** [comment]
`req.body` is expected to be typed as `{ id: string }`. The problem comes from non-validated user input. Static analysis is not enough to detect malicious input present at runtime.

**Comment by jawj** [comment]
But if `req.body` is actually typed as `{ id: string }`, then I think you must at some point have cast it to that without checking it, no?

**Comment by javier-garcia-meteologica** [comment]
> at some point have cast it to that without checking it

Yes, in this issue the developer can definitely be blamed for not validating input. E.g.

```ts
router.post('/example', (req, res, next) => {
  // Junior developer not validating input
  const filter = req.body as { id: string };

  console.log(zp.sql`SELECT * FROM ${'users'} WHERE ${filter}`.compile());
});
```

But `zapatos` could provide a higher level of robustness for free, #99. It's already quoting identifiers, so why not escape them?

My fear is that junior developers get comfortable with typescript very quickly and they think that runtime values are always constrained to types, but that assumption is dangerous when it comes to security.
